### ISSUE TITLE
ipa2tokens should thrown an error when being past something else but a string

### ISSUE THREAD

**Issue by LinguList** [lingpy/lingpy#443]
This should be fixed, as we currently silently allow to handle lists, which does not make sense.

**Comment by LinguList** [comment]
This follows up on #442

**Comment by PatrickElmer** [comment]
#442 fixes this. Passing anything other than a `str` into `ipa2tokens` now raises a value error.

Looking at the code of the `ipa2tokens` function I have realized that if pre-tokenized strings are passed in, it strips any word-initial or word-final #. This is not done for strings that are not pre-tokenized.

```python
>>> ipa2tokens('# a b #')
['a', 'b']
>>> ipa2tokens('#ab#')
['#', 'a', 'b', '#']
```

Should this behavior be removed for pre-tokenized strings or extended to not pre-tokenized strings?

Another small point I realized is that the `ipa2tokens` functions uses the variable name `istring`, while the docstring says `seq`. Which of the two is preferred?

**Comment by xrotwang** [comment]
It could be argued that both behaviours are ok/intended. But in general, I'd find it much more consistent if we treated lists of tokens **always** as Python `list`s, and **not** as whitespace separated tokens in a `str`. (Incidentally, I think, the `istring` / `seq` inconsistency is another symptom of this lack of rigour :) )

**Comment by LinguList** [comment]
We are facing issues from the past here, which could be modified and just ignored. For my first post-doc, I was asked to allow the behavior of always treating `#` in beginning and end of a string as a marker for begin and end, and ignore it. So I added this behavior without really needing it. Since this has long been superseded by CLDF, it is something we can drop from new versions of lingpy.

**Comment by LinguList** [comment]
So this means, we should no longer remove any `#` from the string that is supposed to be segmented in the ipa2tokens function.

**Comment by LinguList** [comment]
Another relic is the behavior of ipa2tokens to return `string.split(" ")` when being passed a string with spaces. This behavior is again something we do not need anymore. Nowadays, and I'd recommend to you, @PatrickElmer, to also switch to this, we segment explicitly our wordlists using orthography profiles, as also described in our more recent tutorials, and we thus do no longer need ipa2tokens on the fly, but only when creating these orthography profiles which we later refine.

**Comment by LinguList** [comment]
That means, more consistent would be to throw an error if you pass ipa2tokens anything that has already as space, arguing that the function would require spaces to be rendered by underscores, as we do in our orthography profiles.

**Comment by LinguList** [comment]
Regarding the documentation showing seq vs. the internal variable name being istring: this is something that can be easily fixed but is probably not really important, as it does not change any of the behaviors and is just a documentation issue.

So to summarize, my suggestion would be:

1. discard the special treatment of `#` as it is no longer needed
2. do not segment strings with spaces, but raise an error, to be more explicit
3. correct the documentation

If I remember correctly, modifying these should not break too much of the current code base.

**Comment by PatrickElmer** [comment]
Thank you for your comments. I have tried to implement a fix in PR #447.

The `msa2html` function internally passes space-separated strings into `ipa2tokens`. This is because the `_list2msa` function loads `["seqs"]` as space-separated strings. The PR changes this to loading `["seqs"]` as a list of tokens.

I am not sure whether this is the right approach for a fix, as I am uncertain if `msa["seqs"]` is expected to be a space-separated string for other use cases.
